py/server/deephaven/experimental/s3.py

```diff
@@ -0,0 +1,63 @@
+#
+# Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
+#
+"""
+    This module is useful for reading and writing files stored in S3. Importing this module requires the S3 specific
+    extensions to be included in the class path.
+"""
+
+import jpy
+
+_JDuration = jpy.get_type("java.time.Duration")
+_JS3Instructions = jpy.get_type("io.deephaven.extensions.s3.S3Instructions")
+
+def _build_s3_instructions(
+        aws_region_name: str,
+        max_concurrent_requests: int = None,
+        read_ahead_count: int = None,
+        fragment_size: int = None,
+        max_cache_size: int = None,
+        connection_timeout: _JDuration = None,
+        read_timeout: _JDuration = None,
```

Comment:
In python, we can support far more time types as input args if this ends up being user facing.  See `to_j_duration` and the methods that use it.